pymc3/variational/opvi.py

```diff
@@ -473,353 +480,165 @@ def from_function(cls, f):
-        model = modelcontext(model)
-        self._scale_cost_to_minibatch = theano.shared(np.int8(0))
-        self.scale_cost_to_minibatch = scale_cost_to_minibatch
-        if not isinstance(cost_part_grad_scale, theano.Variable):
-            self.cost_part_grad_scale = theano.shared(pm.floatX(cost_part_grad_scale))
-        else:
-            self.cost_part_grad_scale = pm.floatX(cost_part_grad_scale)
-        self._seed = random_seed
+    initial_dist_name = 'normal'
+    initial_dist_map = 0.
+
+    def __init__(self, group=None,
+                 shapes=None,
+                 params=None,
+                 size=None,
+                 random_seed=None,
+                 model=None):
         self._rng = tt_rng(random_seed)
+        model = modelcontext(model)
         self.model = model
-        self.check_model(model, **kwargs)
-        if local_rv is None:
-            local_rv = {}
-
-        def get_transformed(v):
-            if hasattr(v, 'transformed'):
-                return v.transformed
-            return v
-
-        known = {get_transformed(k): v for k, v in local_rv.items()}
-        self.known = known
-        self.local_vars = self.get_local_vars(**kwargs)
-        self.global_vars = self.get_global_vars(**kwargs)
-        self._g_order = ArrayOrdering(self.global_vars)
-        self._l_order = ArrayOrdering(self.local_vars)
-        self.gbij = DictToArrayBijection(self._g_order, {})
-        self.lbij = DictToArrayBijection(self._l_order, {})
-        self.symbolic_initial_local_matrix = tt.matrix(self.__class__.__name__ + '_symbolic_initial_local_matrix')
-        self.symbolic_initial_global_matrix = tt.matrix(self.__class__.__name__ + '_symbolic_initial_global_matrix')
-
-        self.global_flat_view = model.flatten(
-            vars=self.global_vars,
-            order=self._g_order,
-        )
-        self.local_flat_view = model.flatten(
-            vars=self.local_vars,
-            order=self._l_order,
-        )
-        self.symbolic_n_samples = self.symbolic_initial_global_matrix.shape[0]
-
-    _global_view = property(lambda self: self.global_flat_view.view)
-    _local_view = property(lambda self: self.local_flat_view.view)
-    local_input = property(lambda self: self.local_flat_view.input)
-    global_input = property(lambda self: self.global_flat_view.input)
-
-    local_names = property(lambda self: tuple(v.name for v in self.local_vars))
-    global_names = property(lambda self: tuple(v.name for v in self.global_vars))
+        if group is None:
+            self.group = model.vars
+        elif group == -1:
+            self.group = -1
+        else:
+            self.group = group
+        if shapes is None:
+            shapes = dict()
+        if params is None:
+            params = dict()
+        self.shapes = shapes
+        self.user_params = params
+        self._mid_size = size
+        self.vmap = dict()
+        self.ndim = 0
+        self._global_ = True
```

Comment:
I see. Another option would be to use all caps (e.g. `self.GLOBAL`). This is common for fixed variables, such as constants in Python. Not sure what PEP8 has to say about that, but the Google style guide uses this convention.